client/signup/steps/page-picker/use-cart-for-difm.tsx

```diff
@@ -0,0 +1,251 @@
+import { buildDIFMCartExtrasObject } from 'calypso/state/difm/assemblers';
+import { requestProductsList } from 'calypso/state/products-list/actions';
+import { getProductBySlug } from 'calypso/state/products-list/selectors';
+import { getSignupDependencyStore } from 'calypso/state/signup/dependency-store/selectors';
+import { fetchSitePlans } from 'calypso/state/sites/plans/actions';
+import { getCurrentPlan } from 'calypso/state/sites/plans/selectors';
+import { isCurrentPlanPaid } from 'calypso/state/sites/selectors';
+import type {
+	MinimalRequestCartProduct,
+	ResponseCart,
+	ResponseCartProduct,
+} from '@automattic/shopping-cart';
+import type { ProductListItem } from 'calypso/state/products-list/selectors/get-products-list';
+import type { TranslateOptions, TranslateResult } from 'i18n-calypso';
+
+export type CartItem = {
+	nameOverride?: TranslateResult;
+	product:
+		| ( Partial< ProductListItem > & Pick< ProductListItem, 'cost' | 'product_name' > )
+		| ( Partial< ResponseCartProduct > & Pick< ResponseCartProduct, 'cost' | 'product_name' > );
+	meta?: TranslateResult;
+	productCount?: number;
+	lineCost: number;
+};
+
+const FREE_PAGES = 5;
+
+type TranslateType = ( original: string, options?: TranslateOptions ) => React.ReactChild;
+type DummyCartParams = {
+	selectedPages: string[];
+	currencyCode: string;
+	isPaidPlan: boolean | null;
+	activePlanScheme: ProductListItem | null;
+	difmLiteProduct: ProductListItem | null;
+	extraPageProduct: ProductListItem | null;
+	translate: TranslateType;
+	formatCurrency: ( original: number, currrency: string ) => string | null;
+};
+
+function getDummyCartProducts( {
+	selectedPages,
+	currencyCode,
+	isPaidPlan,
+	activePlanScheme,
+	difmLiteProduct,
+	extraPageProduct,
+	translate,
+	formatCurrency,
+}: DummyCartParams ): CartItem[] {
+	const extraPageCount = Math.max( 0, selectedPages.length - FREE_PAGES );
+	let displayedCartItems: CartItem[] = [];
+	if ( difmLiteProduct && activePlanScheme && extraPageProduct ) {
+		displayedCartItems = [
+			{
+				nameOverride: translate( 'Website Design Service' ),
+				product: difmLiteProduct,
+				lineCost: difmLiteProduct.cost,
+				meta: translate( 'One-time fee' ),
+			},
+			{
+				product: activePlanScheme,
+				meta: translate( 'Plan Subscription: %(planPrice)s per year', {
+					args: { planPrice: formatCurrency( activePlanScheme.cost, currencyCode ) },
+				} ),
+				lineCost: activePlanScheme.cost,
+			},
+
+			{
+				nameOverride: `${ extraPageCount } ${
+					extraPageCount === 1 ? translate( 'Extra Page' ) : translate( 'Extra Pages' )
+				}`,
+				product: extraPageProduct,
+				meta: translate( '%(perPageCost)s Per Page', {
+					args: {
+						perPageCost: extraPageProduct.cost_display,
+					},
+				} ),
+				lineCost: extraPageProduct.cost * extraPageCount,
+				productCount: extraPageCount,
+			},
+		];
+	}
+
+	// Hide pro plan if existing site has a paid plan
+	if ( activePlanScheme && isPaidPlan ) {
```

Comment:
Since this function will only be called for new sites, we can remove this condition. Extending this, I think 
 the `isPaidPlan` won't be required at all. Wdyt?

Reply:
Agree! 👍